### ISSUE TITLE
signcolumn="yes" + cursorline results in line being visually indented backwards when going from unindented line to indented line

### ISSUE THREAD

**Issue by sqrtnull** [neovim/neovim#29261]
### Problem

![CleanShot 2024-06-10 at 05 34 49@2x](https://github.com/neovim/neovim/assets/51586292/e0c95a03-7fc8-432e-b50f-5ca96e27648a)
this is an image of it occurring and creating artifacts. I managed to reproduce it on version 0.10.0 with only these two options
```
vim.opt.cursorline=true
vim.opt.signcolumn="yes"
```
it is a problem when sshing into remote (supercomputer). I think I sometimes get it locally but I can not reproduce it so it may not be true.
when it occurs with vsplit, things from the right hand side fall over to the left.

![CleanShot 2024-06-10 at 19 54 16](https://github.com/neovim/neovim/assets/51586292/f62fbeab-ae9f-47c3-89f3-c5e6ee47a64b)

### Steps to reproduce

ssh into TSUBAME4 https://www.t4.gsic.titech.ac.jp/docs/handbook.en/
go into an interactive session
qrsh -l cpu_4 -l h_rt=0:10:00

nvim --clean -u minimal.lua

minimal.lua:
```
vim.opt.cursorline=true
vim.opt.signcolumn="yes"
```

check by going from unindented line to indented line
example file:
```

		example
```

### Expected behavior

line does not indent backwards

### Neovim version (nvim -v)

NVIM v0.10.0 Build type: Release LuaJIT 2.1.1713484068

### Vim (not Nvim) behaves the same?

no

### Operating system/version

ssh into remote so idk uname -ra gives Linux r2n6 5.14.0-362.18.1.el9_3.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Jan 3 15:54:45 EST 2024 x86_64 x86_64 x86_64 GNU/Linux

### Terminal name/version

alacritty 0.13.2 (1)

### $TERM environment variable

xterm-256color

### Installation

appimage

**Comment by zeertzjq** [comment]
Cannot reproduce. Does it happen if you set `$TERM` to `alacritty`?

**Comment by sqrtnull** [comment]
sorry I forgot to add that I can only reproduce this on TSUBAME4 https://www.t4.gsic.titech.ac.jp/docs/handbook.en/ and after some testing, it seems to occur when using an interactive session (not the login node) although I cannot say for certain because I cannot get hold of a full node (node_f) right now. Setting the $TERM to alacritty did not fix this.

**Comment by zeertzjq** [comment]
This may have been fixed by #29718. Can you check if the latest nightly release (or v0.10.1 when it comes out) behaves correctly?

**Comment by sqrtnull** [comment]
closing since this is likely a problem with TSUBAME.
